### PR TITLE
Implement ProductImages component

### PR DESCRIPTION
**Pull request by klzns**
[Demo do ProductImages - Watch Video](https://www.loom.com/share/f34c286eb8534ebb88a0c48759624267)

[](https://www.loom.com/share/f34c286eb8534ebb88a0c48759624267)

(I recorded the Loom before updating the repo, that's why the store looks different)

I had to do some changes to implement this component:
- Added a new mode for the Modal called `full`
- Forwarded the prop `onClose` to the `<dialog>` tag of Modal
- Added a new animation called `fade-in`

Change the images array to this to have a better idea of how the component works:

```js
[
  {
    "@type": "ImageObject",
    url:
      "https://store.storeimages.cdn-apple.com/4982/as-images.apple.com/is/MMMQ3?wid=1144&hei=1144&fmt=jpeg&qlt=90&.v=1645138486301",
    alternateName: "Magic Mouse",
  },
  {
    "@type": "ImageObject",
    url:
      "https://store.storeimages.cdn-apple.com/4982/as-images.apple.com/is/MMMQ3_AV1?wid=1144&hei=1144&fmt=jpeg&qlt=90&.v=1645138490227",
    alternateName: "Magic Mouse",
  },
  {
    "@type": "ImageObject",
    url:
      "https://store.storeimages.cdn-apple.com/4982/as-images.apple.com/is/MMMQ3_AV2?wid=1144&hei=1144&fmt=jpeg&qlt=90&.v=1645138490081",
    alternateName: "Magic Mouse",
  },
  {
    "@type": "ImageObject",
    url:
      "https://store.storeimages.cdn-apple.com/4982/as-images.apple.com/is/MMMQ3_AV3?wid=1144&hei=1144&fmt=jpeg&qlt=90&.v=1645138485700",
    alternateName: "Magic Mouse",
  },
  {
    "@type": "ImageObject",
    url:
      "https://store.storeimages.cdn-apple.com/4982/as-images.apple.com/is/MMMQ3_AV4?wid=1144&hei=1144&fmt=jpeg&qlt=90&.v=1645138484328",
    alternateName: "Magic Mouse",
  },
]
```

### CODE REVIEW

File: components/product/ProductImages/index.tsx [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+export type Props = {
+  image: ImageObject[];
+  width: number;
+  height: number;
+  thumbWidth: number;
+  thumbHeight: number;
+};
+
+export function ProductImages(
```
Comment (igorbrasileiro) [inline components/product/ProductImages/index.tsx:15]: By doing this:
```suggestion
export default function ProductImages(
```

the island can be somehting like this:
```ts
export {default} from "$store/components/product/ProductImages/index.ts"
export type {Props} from "$store/components/product/ProductImages/index.ts"
```

**Comment by guifromrio** [comment]
Amazing!! A "reference PR"!

File: components/product/ProductImages/index.tsx [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      <ProductImageZoom
+        id={id}
+        image={image}
+        width={width}
+        height={height}
+        currentImage={currentImage}
+        setCurrentImage={setCurrentImage}
+        open={isZoomOpen}
+        setZoomOpen={setZoomOpen}
+      />
```
Comment (igorbrasileiro) [inline components/product/ProductImages/index.tsx:76]: Wdyt about lazying load this ProductImage? I mean, something like this:

```tsx
const ProductImageZoom = lazy(() => import('./ProductImageZoom.tsx'))
// ...

{isZoomOpen && (
<Suspense fallback={null}>
  <ProductImageZoom {...} />
</Suspense>
)}
```

You can does this too https://github.com/deco-sites/fashion/blob/main/components/header/Modals.tsx#L25-L33

File: components/product/ProductImages/ProductImageZoom.tsx [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+            }}
+          />
+
+          {image.map((img, index) => (
+            <div
+              key={index}
+              id={`zoom-${id}-${index}`}
+              role="tabpanel"
+              aria-label={`Item ${index + 1}`}
+              aria-hidden={currentImage === index}
```
Comment (igorbrasileiro) [inline components/product/ProductImages/ProductImageZoom.tsx:53]: should `aria-hidden` be `currentImage !== index`, shouldn't?

**Review by igorbrasileiro — COMMENTED** [review]
LGTM, i left some comments.

Have you thought about doing the thumbnail select html/css only? Ain't sure if it is possible

**Comment by lucis** [comment]
Really cool, @klzns!

Good points brought by @igorbrasileiro

**Review by igorbrasileiro — APPROVED** [review]

**Comment by klzns** [comment]
Closing outdated PR

**Pull request closed without merging**